lib/src/logger.dart

```diff
@@ -27,21 +27,29 @@ class Logger {
 
   /// The full name of this logger, which includes the parent's full name.
   String get fullName =>
-      (parent == null || parent.name == '') ? name : '${parent.fullName}.$name';
+      parent?.name.isNotEmpty ?? false ? '${parent!.fullName}.$name' : name;
 
   /// Parent of this logger in the hierarchy of loggers.
-  final Logger parent;
+  final Logger? parent;
 
   /// Logging [Level] used for entries generated on this logger.
-  Level _level;
+  ///
+  /// Only the root logger is guaranteed to have a non-null [Level].
```

Comment:
Do you know how this guarantee is made? The setter below doesn't look like it would enforce that null isn't assigned to the root logger's `level`...

Reply:
It looks like this is only asserted when retrieving the level https://github.com/dart-lang/logging/pull/82/files#diff-8f5e033ea3a757e19380c9e27ea174adR106.

I can add a check in when setting it though.

Reply:
Done, I made it so you can't set it to null on any logger without a parent, since we assert that in the getter.